File: 56-merge-intervals/56-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> (a[0] - b[0]));
        List<int []> ans = new ArrayList<>();
        ans.add(intervals[0]);
        int ansSize = 1;
        for (int i = 1; i < intervals.length; i++){
            int s = intervals[i][0];
            int e = intervals[i][1];
            if(ans.get(ansSize - 1)[1] >= intervals[i][0]){
                ans.get(ansSize - 1)[1] = Math.max(ans.get(ansSize-1)[1], intervals[i][1]);
            }
            else{
                ans.add(intervals[i]);
                ansSize++;
            }
        }
        
        return ans.toArray(new int[0][]);
    }
}

File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key = lambda x: int(x[0]))
        res = []
        prev = intervals[0]
        for i in range(1, len(intervals)):
            curr = intervals[i]
            if curr[0] <= prev[1]:
                prev[1] = max(prev[1], curr[1])
            else:
                res.append(prev)
                prev = curr
        
        res.append(prev)
        return res
        
        
            